native/Avalonia.Native/src/OSX/window.mm

```diff
@@ -2047,30 +2064,29 @@ - (void)windowWillClose:(NSNotification *)notification
 
 -(BOOL)canBecomeKeyWindow
 {
-    return _canBecomeKeyAndMain;
+    if (_canBecomeKeyAndMain)
+    {
+        // If the window has a child window being shown as a dialog then don't allow it to become the key window.
+        for(NSWindow* uch in [self childWindows])
+        {
+            auto ch = objc_cast<AvnWindow>(uch);
+            if(ch == nil)
+                continue;
+            if (ch.isDialog)
+                return false;
+        }
+        
+        return true;
+    }
+    
+    return false;
 }
 
 -(BOOL)canBecomeMainWindow
 {
     return _canBecomeKeyAndMain;
```

Comment:
As I see, it should be renamed to "_canBecomeMain"

Reply:
Yeah, I'm not 100% sure why this flag is needed as there's no explanation for it whatsoever. However it's still used in `canBecomeKeyWindow` in that the result from that method is `_canBecomeKeyAndMain && [HasNoDialogChild]` so it made sense to me to keep the name as it still affects whether the window can become a key window.